File: estudante.py
import unittest

class TestEstudante(unittest.TestCase):
    def test_calculo_media(self):
        estudante = Estudante("Carlos", [8, 7, 9, 6])
        self.assertEqual(estudante.calcular_media(), 7.5)

if _name_ == "_main_":
    unittest.main()

class Estudante:
    def _init_(self, nome, notas):
        self.nome = nome
        self.notas = notas

    def calcular_media(self):
        return sum(self.notas) / len(self.notas)

File: estudante.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class EstudanteTest {

    @Test
    void testCalcularMedia() {
        Estudante estudante = new Estudante("Carlos", new double[]{8, 7, 9, 6});
        assertEquals(7.5, estudante.calcularMedia(), 0.01);
    }
}

public class Estudante {
    private String nome;
    private double[] notas;

    public Estudante(String nome, double[] notas) {
        this.nome = nome;
        this.notas = notas;
    }

    public double calcularMedia() {
        double soma = 0;
        for (double nota : notas) {
            soma += nota;
        }
        return soma / notas.length;
    }

    public String getNome() {
        return nome;
    }

    public double[] getNotas() {
        return notas;
    }
}